### ISSUE TITLE
Readonly state is not being conveyed when using a screen reader.

### ISSUE THREAD

**Issue by jake-costa** [jpmorganchase/salt-ds#4586]
### Latest version

- [x] I have tested the latest version

### Description

Readonly state is not being conveyed when using a screen reader.

Effected Components:
Combobox
Dropdown
Radio button
Stepper Input

### Steps to reproduce

.

### Package name(s)

Core (@salt-ds/core), Lab (@salt-ds/lab)

### Browser

- [x] Chrome
- [x] Safari
- [x] Firefox
- [ ] Microsoft Edge

### Operating system

- [x] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.

**Comment by joshwooding** [comment]
Just confirming, this is a browser/screen reader limitation right? I've tested a few other implementations and they all have the same issue?

**Comment by mark-tate** [comment]
Updated documentation with the suggested workaround,  is the way forward.
Note: W3C recommend against using both read-only and disabled.

Will be very hard to remove our own baked-in solution, once the browser's support this, hence a documented workaround that products can remove themselves, is the preferred approach.

**Comment by jake-costa** [comment]
@joshwooding 

From what I found, The readonly HTML attribute is only allowed on text fields (input and textarea), so the readonly attribute does not apply to elements like role=(combobox, radiogroup, listbox, etc). And while we could swap it with aria-readonly which is meant for these non-native implementations, like we have for dropdown, support for aria-readonly is nearly non-existent.

Since the user us unable to edit these, it might be more ideal and make more sense to remove the role and maybe set type="text" for each of these components when in a readonly state so that it is conveyed to screen readers correctly when using HTML readonly. This way we shouldn't be impacted if browsers ever end up supporting this. Thoughts?

Regardless, I'd like to connect when you have some time to discuss this as well as what a valid workaround that could be implemented by product would be.

Resources:
https://github.com/w3c/html-aria/issues/384
https://github.com/FreedomScientific/standards-support/issues/481
https://github.com/FreedomScientific/standards-support/issues/592
https://html.spec.whatwg.org/dev/input.html#attr-input-readonly

**Comment by joshwooding** [comment]
@jake-costa Do you think that would be a work around we would support or a suggestion to our users. Ultimately the question becomes if read-only being announced is more important than the type of control

**Comment by mark-tate** [comment]
update required

**Comment by mark-tate** [comment]
PR open - to be reviewed by EOS
